lib/puppet/provider/f5_sslkey/rest.rb

```diff
@@ -0,0 +1,41 @@
+require 'puppet/provider/f5'
+require 'json'
+
+Puppet::Type.type(:f5_sslkey).provide(:rest, parent: Puppet::Provider::F5) do
+
+  def create_message(basename, hash)
+    # Create the message by stripping :present.
+    new_hash            = hash.reject { |k, _| [:ensure,:name, :provider, Puppet::Type.metaparams].flatten.include?(k) }
+  #  new_hash[:name]     = basename
+
+    return new_hash
+  end
+
+  def message(object)
+    # Allows us to pass in resources and get all the attributes out
+    # in the form of a hash.
+    message = object.to_hash
+
+    # Map for conversion in the message.
+    map = {
+    }
+
+   message = { "command"=>"install", "name"=> message[:keyname], "from-local-file"=> message[:from_local_file] }
+   message.to_json
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present
```

Comment:
If you don't implement `instances`, `@property_hash` will never be set.

Reply:
You are right. In this case, it does not need to.  @property_hash[:ensure] not being set, will trigger "create". @abrader

Reply:
on every single run...

Reply:
@ericzji @alexjfisher absolutely agreed.

Reply:
Modified " @property_hash[:ensure] == :present" to "false", and it's tested